creator-node/test/asyncRetry.test.js

```diff
@@ -16,8 +16,59 @@ describe('test asyncRetry', function () {
         minTimeout: 0,
         maxTimeout: 100
       })
+      assert.fail('Should have thrown')
     } catch (e) {
       assert.deepStrictEqual(e.message, 'Will not retry after 3 retries')
     }
   })
+
+  it('on certain retry, do something', async function () {
+    try {
+      let name = ''
+      await asyncRetry({
+        asyncFn: async (bail, num) => {
+          if (num === 3) {
+            name = 'vicky'
+            return
+          }
+
+          name = 'not vicky'
+          throw new Error(`Test ${num}`)
+        },
+        options: {
+          minTimeout: 0,
+          maxTimeout: 100
+        }
+      })
+
+      assert.deepStrictEqual(name, 'vicky')
+    } catch (e) {
+      assert.fail('Should not have thrown')
+    }
+  })
+
+  it('final error message should be what is specified in num retry check', async function () {
+    try {
+      const retries = 3
+      await asyncRetry({
+        asyncFn: async (bail, num) => {
+          console.log('Test num', num)
```

Comment:
is this still needed?

Reply:
yeah, I figure it'd be an easy example for anyone who needs to implement bail to recreate in their use cases